packages/client/src/pages/FindAMentor/index.js

```diff
@@ -17,21 +39,38 @@ const FindAMentorPage = () => {
 
     const { user } = useAuth();
     const { tags, setTags } = useState([]);
-    const { subjects, setSubjects} = useState([]);
+    const { subjects, setSubjects } = useState([]);
+
+    // Lazily initialize the filter settings
+    const [filters, setFilters] = useState({
+        'gradeLevel': {},
+        'subject': {},
+        'specialNeeds': {},
+    });
     const handleMentorshipRequest = (event) => {
         // TODO(johanc): Implement.
-        
+
     };
 
     const openModal = (mentor) => {
         // TODO(johanc): Implement.
     }
-    const FilterColumn = () => {
-        // TODO: Implement - should use setTags() to update the state.
-        <a onClick={() => setTags([])}>Placeholder filter component.</a>
-    };
 
-    const mentors = await get("/api/mentors", {tags: tags, subjects: subjects});
+    const handleFilterChange = (filterCategory, change) => {
+        console.log(filterCategory, change);
+        if (filters[filterCategory]) {
+            const currentCategoryFilters = filters[filterCategory];
+            setFilters({
+                ...filters,
+                [filterCategory]: {
+                    ...currentCategoryFilters,
+                    [change.name]: change.checked
+                }
+            })
```

Comment:
why do you need brackets around `filterCategory` and `change.name`?

Reply:
this is just shorthand that lets me change keys in the object based on the variable (filterCategory and change.name)

Reply:
oh I see makes sense